### ISSUE TITLE
Extra line on Ctrl + C on 2 line prompt (low priority)

### ISSUE THREAD

**Issue by TheToto** [IlanCosman/tide#53]
#### Describe the bug

When I press Ctrl + C on a empty prompt, it simply redraw the prompt.
When I press Ctrl + C on a prompt with some text, it redraw the prompt with a extra unwanted line (see screenshots)

#### Steps to reproduce

1. Use a 2 line prompt
2. Type something
3. Press Ctrl + C

#### Expected behavior

A clean redraw

#### Screenshots

Before Ctrl + C
![image](https://user-images.githubusercontent.com/12860648/101982741-3a7cd880-3c76-11eb-8cae-1cb53f5f426e.png)
After Ctrl + C
![image](https://user-images.githubusercontent.com/12860648/101982744-3ea8f600-3c76-11eb-988a-2b3f6410c67f.png)

#### Environment (please complete the following information)

- Operating System:  ArchLinux
- Output of `tide bug-report`: 

```console
fish, version 3.1.2
tide, version 2.5.0
xterm-256color

```

**Comment by IlanCosman** [comment]
I believe this is an upstream bug. The same issue will appear with any multi-line prompt:

```fish
function fish_prompt
    echo foo
    echo bar
end
```

**Comment by IlanCosman** [comment]
I'm removing upstream because this isn't really a bug from fish's perspective. I found a way to work around fish's behavior, so this issue will be fixed after the 3.0 release.

**Comment by jorgebucaran** [comment]
Does that mean we can bring back the "But Fish can't do that yet, right?" 🙏

**Comment by IlanCosman** [comment]
It seems I messed up again here 😂 This *was* a bug from the fish maintainer's perspective: https://github.com/fish-shell/fish-shell/issues/3537. It's fixed for the 3.2 release.

Even more strangely, is that fish 3.1.2 behaves differently on MacOS and Linux. On MacOS, things seem to be working just fine for multi-line prompt cancellation, while on Linux we see the problem this issue described. Thus, the changes I made for this issue fix things on Linux, but then make MacOS bug 🤦‍♂️

So I've reverted the changes meant to fix this problem. Rules for next time: check upstream first, and test both OSes.
